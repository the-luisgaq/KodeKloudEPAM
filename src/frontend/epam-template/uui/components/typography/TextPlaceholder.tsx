import * as React from 'react';
import cx from 'classnames';
import { IHasCX, IHasRawProps } from '@epam/uui-core';
import css from './TextPlaceholder.module.scss';
import { PropsWithChildren } from 'react';

export interface ITextPlaceholderProps extends IHasRawProps<React.HTMLAttributes<HTMLDivElement>>, IHasCX {
    /** Number of placeholder worlds to render in block */
    wordsCount?: number;
    /** Pass true, to turn off the placeholder animation */
    isNotAnimated?: boolean;
}

export type TextPlaceholderProps = PropsWithChildren<ITextPlaceholderProps>;

export const TextPlaceholder: React.FunctionComponent<PropsWithChildren<ITextPlaceholderProps>> = (props) => {
    const pattern = '&nbsp;';
    const text = React.useMemo(() => {
        const words = [];
        for (let i = 0; i < (props.wordsCount || 1); i++) {
            const lengthWord = Math.floor(Math.random() * 10 + 8);
            words.push(pattern.repeat(lengthWord));
        }
        return words;
    }, [props.wordsCount]);

    return (
        <div aria-busy={ true } className={ cx(css.root, 'uui-text-placeholder') } { ...props.rawProps }>
            {text.map((it: string, index: number) => (
                <span
                    key={ index }
                    className={ cx([
                        props.cx, css.loadingWord, !props.isNotAnimated && css.animatedLoading,
                    ]) }
                    dangerouslySetInnerHTML={ { __html: it } }
                />
            ))}
        </div>
    );
};
