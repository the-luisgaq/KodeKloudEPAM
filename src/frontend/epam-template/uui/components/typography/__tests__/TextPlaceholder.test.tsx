import React from 'react';
import { TextPlaceholder } from '../TextPlaceholder';
import { render } from '@epam/uui-test-utils';

describe('TextPlaceholder', () => {
    beforeEach(() => {
        jest.spyOn(global.Math, 'random').mockReturnValue(0.5);
    });

    afterEach(() => {
        jest.spyOn(global.Math, 'random').mockRestore();
    });

    it('should be rendered correctly', () => {
        const { asFragment } = render(<TextPlaceholder />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be rendered correctly', () => {
        const { asFragment } = render(<TextPlaceholder wordsCount={ 5 } isNotAnimated />);
        expect(asFragment()).toMatchSnapshot();
    });
});
