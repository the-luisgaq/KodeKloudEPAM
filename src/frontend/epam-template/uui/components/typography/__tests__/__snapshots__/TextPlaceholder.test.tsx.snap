// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextPlaceholder should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    class="root uui-text-placeholder"
  >
    <span
      class="loadingWord animatedLoading"
    >
                   
    </span>
  </div>
</DocumentFragment>
`;

exports[`TextPlaceholder should be rendered correctly 2`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    class="root uui-text-placeholder"
  >
    <span
      class="loadingWord"
    >
                   
    </span>
    <span
      class="loadingWord"
    >
                   
    </span>
    <span
      class="loadingWord"
    >
                   
    </span>
    <span
      class="loadingWord"
    >
                   
    </span>
    <span
      class="loadingWord"
    >
                   
    </span>
  </div>
</DocumentFragment>
`;
